ocaml/xapi-storage-script/main.ml

```diff
@@ -101,30 +357,61 @@ let return_plugin_rpc result = return_rpc Xapi_storage.Common.typ_of_exnt result
 
 let return_data_rpc result = return_rpc Xapi_storage.Common.typ_of_exnt result
 
-let use_syslog = ref false
-
-let log level fmt =
-  Printf.ksprintf
-    (fun s ->
-      let module Writer = Async.Writer in
-      if !use_syslog then
-        (* FIXME: this is synchronous and will block other I/O.
-         * This should use Log_extended.Syslog, but that brings in Core's Syslog module
-         * which conflicts with ours *)
-        Syslog.log Syslog.Daemon level s
-      else
-        let w = Lazy.force Writer.stderr in
-        Writer.write w s ; Writer.newline w
+(* Reporter taken from
```

Comment:
Do we still need to copy/paste this into every Lwt program?
Perhaps we should have a small function in a common library somewhere in XAPI that does this, there are other Lwt programs too: wsproxy/xapi-guard/xapi-nbd, etc.

Reply:
We probably should add this to the xapi-logs library, maybe as a sublibrary so not everything links against lwt